engine/classes/ElggFile.php

```diff
@@ -469,4 +469,66 @@ public function __sleep() {
 		]);
 	}
 
+	/**
+	 * Checks the download permissions for the file
+	 * 
+	 * @param int  $user_guid GUID of the user (defaults to logged in user)
+	 * @param bool $default   Default permission
+	 */
+	public function canDownload($user_guid = 0, $default = true) {
+		 return _elgg_services()->userCapabilities->canDownload($this, $user_guid, $default);
+	}
+	
+	/**
+	 * Returns file's download URL
+	 *
+	 * @note This does not work for files with custom filestores.
+	 *
+	 * @param bool    $use_cookie Limit URL validity to current session only
+	 * @param string  $expires    URL expiration, as a string suitable for strtotime()
+	 * @return string
+	 */
+	public function getDownloadURL($use_cookie = true, $expires = '+2 hours') {
+
+		$file_svc = new \Elgg\FileService\File();
+		$file_svc->setFile($this);
+		$file_svc->setExpires($expires);
+		$file_svc->setDisposition('attachment');
+		$file_svc->bindSession($use_cookie);
+		$url = $file_svc->getURL();
+
+		$params = [
+			'entity' => $this,
```

Comment:
No, don't like it. File service is bound to instance of ElggFile and other parameters of the download/inline URL. I'd rather plugins instantiate it again and build their own parameters if that's an intention. My motivation is to fallback to non /serve-file/ URLs if needed. 
You can add the file service later if there is a compelling case and add the appropriate tests.